### PR TITLE
Add `isRbfTransaction` util, improve types

### LINKED ISSUE #13325: Cancelled bump fee modal creates send draft
**Describe the bug**
When I cancel bump fee tx review modal and I navigate to send form, it is partially filled and symbol is missing. I would not expect filled form at all!

If the tx is mined with old fee and user will want to send another tx in send form, it will not work as it adds also rbf params

**Info:**

-   Suite Version: 24.7.1

**How to reproduce**
1. Send tx with low gas price
1. Bump fee
2. Cancel it in review tx modal
3. Navigate to send form

**Screenshots**
Screenshot 2024-07-14 at 20 55 33

### PR DESCRIPTION
**Pull request by komret**
## Description

My intention was to fix https://github.com/trezor/trezor-suite/issues/13325, but it is tricky (see https://github.com/trezor/trezor-suite/issues/13325#issuecomment-2278752746), so I at least commit some refactoring.

The transaction types are a mess, this should improve it a little. 🤯

### CODE REVIEW

File: packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx [hunk 1]
```diff
@@ -71,9 +71,10 @@ export const TransactionReviewModalContent = ({
             : selectSendFormDraftByAccountKey(state, account?.key),
     );
 
-    const decreaseOutputId = precomposedTx?.useNativeRbf
-        ? precomposedForm?.setMaxOutputId
-        : undefined;
+    const isRbfAction = precomposedTx !== undefined && isRbfTransaction(precomposedTx);
+
+    const decreaseOutputId =
+        isRbfAction && precomposedTx?.useNativeRbf ? precomposedForm?.setMaxOutputId : undefined;
```
Comment (marekrjpolak) [inline packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx]: nit: useless optional chaining
```suggestion
        isRbfAction && precomposedTx.useNativeRbf ? precomposedForm?.setMaxOutputId : undefined;
```

File: suite-common/wallet-utils/src/reviewTransactionUtils.ts [hunk 2]
```diff
@@ -181,7 +184,7 @@ const constructOldFlow = ({
                 value: precomposedForm.rippleDestinationTag,
             });
         }
-    } else if (!precomposedTx.useNativeRbf) {
+    } else if (isRbf && !precomposedTx.useNativeRbf) {
```
Comment (marekrjpolak) [inline suite-common/wallet-utils/src/reviewTransactionUtils.ts]: I have to admit I'm totally missing a bigger picture here, but this is different than before, right? This branch used to apply for non-native rbf txs **and** non-rbf txs, now it completely ignores non-rbf txs. Is it wanted?

File: suite-common/wallet-core/src/send/sendFormThunks.ts [hunk 3]
```diff
@@ -538,14 +539,17 @@ export const enhancePrecomposedTransactionThunk = createThunk<
...
+                    formValues.rbfParams.txid;
+                (enhancedPrecomposedTransaction as PrecomposedTransactionFinalRbf).feeDifference =
+                    new BigNumber(precomposedTransaction.fee)
+                        .minus(formValues.rbfParams.baseFee)
+                        .toFixed();
+                (enhancedPrecomposedTransaction as PrecomposedTransactionFinalRbf).useNativeRbf =
+                    !!useNativeRbf;
+                (
+                    enhancedPrecomposedTransaction as PrecomposedTransactionFinalRbf
+                ).useDecreaseOutput = !!hasDecreasedOutput;
```
Comment (marekrjpolak) [inline suite-common/wallet-core/src/send/sendFormThunks.ts:552]: Not exactly beautiful, but I haven't come with anything better, so consider this a pointless rant.

File: packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx (hunk 1 above)
Comment (komret) [reply packages/suite/src/components/suite/modals/ReduxModal/TransactionReviewModal/TransactionReviewModalContent.tsx]: 1aabb1b7515a434e01e2e48366db9635d2f3edf2

File: suite-common/wallet-utils/src/reviewTransactionUtils.ts (hunk 2 above)
Comment (komret) [reply suite-common/wallet-utils/src/reviewTransactionUtils.ts]: It is there to satisfy TypeScript. `useNativeRbf` does not exist on `PrecomposedTransactionCardanoFinal` so I have to use `isRbfTransaction` as a type guard.

File: suite-common/wallet-core/src/send/sendFormThunks.ts (hunk 3 above)
Comment (komret) [reply suite-common/wallet-core/src/send/sendFormThunks.ts:552]: I felt the same.

File: suite-common/wallet-utils/src/reviewTransactionUtils.ts (hunk 2 above)
Comment (marekrjpolak) [reply suite-common/wallet-utils/src/reviewTransactionUtils.ts]: I know, I'm talking about `precomposedTx.useNativeRbf === undefined` case, which satisfied the condition before but not now. Is it something which can happen?

Comment (komret) [reply suite-common/wallet-utils/src/reviewTransactionUtils.ts]: 6774c3f5c385da7dd29e8140f6d4129de04ed7da

**Review by marekrjpolak — APPROVED** [review]

**Pull request merged**